src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -146,9 +146,25 @@ private function parseDefinitions($content, $file)
         if (!is_array($content['services'])) {
             throw new InvalidArgumentException(sprintf('The "services" key should contain an array in %s. Check your YAML syntax.', $file));
         }
+        if (isset($content['services']['_defaults'])) {
+            $defaults = $content['services']['_defaults'];
+            $defaultKeys = array('public', 'tags', 'autowire');
+            unset($content['services']['_defaults']);
+
+            if (!is_array($defaults)) {
+                throw new InvalidArgumentException(sprintf('Service defaults must be an array, "%s" given in "%s".', gettype($defaults), $file));
+            }
```

Comment:
I would validate the default value for ``tags`` here (when defined) instead of letting the first service using it throw an exception. It would be more understandable.